hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/WritableAirdropStore.java

```diff
@@ -0,0 +1,112 @@
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.PendingAirdropId;
+import com.hedera.hapi.node.base.PendingAirdropValue;
+import com.hedera.node.app.spi.metrics.StoreMetricsService;
+import com.hedera.node.config.data.TokensConfig;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.state.spi.WritableKVState;
+import com.swirlds.state.spi.WritableStates;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * Provides write methods for modifying underlying data storage mechanisms for
+ * working with Pending Airdrops.
+ *
+ * <p>This class is not exported from the module. It is an internal implementation detail.
+ * This class is not complete, it will be extended with other methods like remove, update etc.,
+ */
+public class WritableAirdropStore extends ReadableAirdropStoreImpl {
+    /**
+     * The underlying data storage class that holds the Pending Airdrops data.
+     */
+    private final WritableKVState<PendingAirdropId, PendingAirdropValue> airdropState;
+
+    /**
+     * Create a new {@link WritableAirdropStore} instance.
+     *
+     * @param states The state to use.
+     */
+    public WritableAirdropStore(
+            @NonNull final WritableStates states,
+            @NonNull final Configuration configuration,
+            @NonNull final StoreMetricsService storeMetricsService) {
+        super(states);
+        airdropState = states.get(AIRDROPS_KEY);
+
+        final long maxCapacity = configuration.getConfigData(TokensConfig.class).maxAllowedAirdorps();
+        final var storeMetrics = storeMetricsService.get(StoreMetricsService.StoreType.AIRDROPS, maxCapacity);
+        airdropState.setMetrics(storeMetrics);
+    }
+
+    /**
+     * Persists a new {@link PendingAirdropId} with given {@link PendingAirdropValue} into the state,
+     * as well as exporting its ID to the transaction receipt. If there is existing
+     * airdrop with the same id we add the value to the existing drop.
+     *
+     * @param airdropId    - the airdropId to be persisted.
+     * @param airdropValue - the value for the given airdropId to be persisted.
+     */
+    public void put(@NonNull final PendingAirdropId airdropId, @NonNull final PendingAirdropValue airdropValue) {
+        requireNonNull(airdropId);
+        requireNonNull(airdropValue);
+        if (airdropState.contains(airdropId)) {
+            if (airdropId.hasFungibleTokenType()) {
+                var existingAirdropValue =
+                        requireNonNull(airdropState.get(airdropId)).amount();
+                var newValue = airdropValue.amount() + existingAirdropValue;
+                var newAirdropValue =
+                        PendingAirdropValue.newBuilder().amount(newValue).build();
+                airdropState.put(airdropId, newAirdropValue);
+            }
+        } else {
+            airdropState.put(airdropId, airdropValue);
+        }
```

Comment:
Nit: I like having as little nestings as possible - I think it reads better. This is just a suggestion and personal preference if you don't like it you can leave it as is:
```
requireNonNull(airdropId);
        requireNonNull(airdropValue);
        
        if (!airdropState.contains(airdropId)) {
            airdropState.put(airdropId, airdropValue);
            return;
        }
        
        if (!airdropId.hasFungibleTokenType()) {
            return;
        }
        
        var existingAirdropValue = requireNonNull(airdropState.get(airdropId)).amount();
        var newValue = airdropValue.amount() + existingAirdropValue;
        var newAirdropValue = PendingAirdropValue.newBuilder().amount(newValue).build();
        airdropState.put(airdropId, newAirdropValue);
```

Reply:
Rewritten.